mlflow/spark.py

```diff
@@ -98,10 +99,20 @@ def get_default_pip_requirements():
     if Version(pyspark.__version__) < Version("3.4"):
         # Versions of PySpark < 3.4 are incompatible with pandas >= 2
         reqs.append("pandas<2")
+
+    if is_spark_connect_model:
+        reqs.extend([
+            # Spark connect Model uses spark torch distributor to train model
+            _get_pinned_requirement("torch"),
```

Comment:
so we need to install torch and sklearn if we want to use spark connect models?

Reply:
Yes :)

Reply:
What happens if we try to use spark connect models without the required dependencies?

Reply:
Currently, for distributed trained model, it is saved as torch model format, and for feature transformer model, it is saved as sklearn model. Without required dependencies model loading will fail.

For `torcheval`, currently it is only used in cross validation training, I think we can remove it from default pip requirements.

Reply:
Does pyspark has an extra for spark connect?

Reply:
`torcheval` is fine because it's tiny.

Reply:
> Does pyspark has an extra for spark connect?

Yes it has, but currently it does not include ML-connect dependencies.

Reply:
should we use it?

Reply:
We can add an pyspark extra like "ml-connect" to include these dependencies, spark 3.5 is going to release soon, we have to add it into spark 4.0

Reply:
Do spark connect ML models without spark connect dependencies?

Reply:
> Do spark connect ML models work without spark connect dependencies, or not?

It only works with spark connect dependencies installed too.

Reply:
Should we use `pyspark[connect]` instead of `pyspark`?

Reply:
Yes. you are right. :)